server/src/com/treode/server/Resource.scala

```diff
@@ -97,7 +97,30 @@ class Resource (host: HostId, store: SchematicStore) extends Service [Request, R
         respond (req, Status.PreconditionFailed)
     }}
 
+  def batch_write (req: Request): Async [Response] = {
+    val tx = req.transactionId (host)
+    val ct = req.conditionTxClock (TxClock.now)
+    try {
+      store.batch_write(tx, ct, req)
+      .map [Response] { vt =>
+        val rsp = req.response
+        rsp.status = Status.Ok
+        rsp.headerMap.add ("Value-TxClock", vt.toString)
+        rsp
+      }
+      .recover {
+        case _: StaleException => {
+          respond (req, Status.PreconditionFailed)
+        }
+      }
+    } catch {
+      case v: Throwable => {
```

Comment:
In this context, we must not catch all Throwables and write them to the client. Doing so could leak something secret. Who knows what, it's a just a simple precaution that we take: we only expose Exceptions that have been blessed for that purpose. How to bless an Exception as exposable? Make it extend `BadRequestException`. Once you do that, you no longer need to deliberately catch it and translate it to an HTTP response. There's a filter that does that for you in place already.